src/Compilers/Core/Portable/SourceGeneration/IncrementalGeneratorRunStep.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// A description of a step of an incremental generator that was executed.
+    /// </summary>
+    public class IncrementalGeneratorRunStep
```

Comment:
struct?

Reply:
I prefer having the step type be a reference type so that any operations that construct a graph of the steps can use reference equality to easily determine when a step's output is used in two different dependent steps.